nixos/modules/services/mail/tlsrpt.nix

```
@@ -0,0 +1,324 @@
+                config.services.tlsrpt.collectd.settings.storage
+              '';
+              description = ''
+                Path to the collectd sqlite database.
+              '';
+            };
+
+            log_level = mkOption {
+              type = types.enum [
+                "debug"
+                "info"
+                "warning"
+                "error"
+                "critical"
+              ];
+              default = "info";
+              description = ''
+                Level of log messages to emit.
+              '';
+            };
+          };
+        };
+        default = { };
+        description = ''
+          Flags from {manpage}`tlsrpt-fetcher(1)` as key-value pairs.
+        '';
+      };
+    };
+
+    reportd = {
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = format.type;
+          options = {
+            dbname = mkOption {
+              type = types.str;
+              default = "/var/lib/tlsrpt/reportd.sqlite";
+              description = ''
+                Path to the sqlite database.
+              '';
+            };
+
+            fetchers = mkOption {
+              type = types.str;
+              default = lib.getExe' cfg.package "tlsrpt-fetcher";
+              defaultText = lib.literalExpression ''
+                lib.getExe' cfg.package "tlsrpt-fetcher"
+              '';
+              description = ''
+                Comma-separated list of fetcher programs that retrieve collectd data.
+              '';
+            };
+
+            log_level = mkOption {
+              type = types.enum [
+                "debug"
+                "info"
+                "warning"
+                "error"
+                "critical"
+              ];
+              default = "info";
+              description = ''
+                Level of log messages to emit.
+              '';
+            };
+
+            organization_name = mkOption {
+              type = types.str;
+              example = "ACME Corp.";
+              description = ''
+                Name of the organization sending out the reports.
+              '';
+            };
+
+            contact_info = mkOption {
+              type = types.str;
+              example = "smtp-tls-reporting@example.com";
+              description = ''
+                Contact information embedded into the reports.
+              '';
+            };
+
+            sender_address = mkOption {
+              type = types.str;
+              example = "noreply@example.com";
+              description = ''
+                Sender address used for reports.
+              '';
+            };
+
+            sendmail_script = mkOption {
+              type = with types; nullOr str;
+              default = if config.services.postfix.enable then "sendmail" else null;
+              defaultText = lib.literalExpression ''
+                if any [ config.services.postfix.enable ] then "sendmail" else null
```

```suggestion
                if config.services.postfix.enable then "sendmail" else null
```
that's matching default

---

I don't know why you marked this as resolved.

---

I do wonder where the hunk went, because I made this change locally4.